ydkgen/printer/cpp/entity_lookup_printer.py

```diff
@@ -0,0 +1,93 @@
+#  ----------------------------------------------------------------
+# Copyright 2016 Cisco Systems
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------
+
+"""
+entity_lookup_printer.py
+
+ prints top entity lookup map
+
+"""
+from ydkgen.printer.file_printer import FilePrinter
+
+
+class EntityLookUpPrinter(FilePrinter):
+    def __init__(self, ctx):
+        super(EntityLookUpPrinter, self).__init__(ctx)
+        self.headers = None
+        self.entity_lookup = None
+
+    def print_entity_lookup(self, packages, sort_clazz):
+        if sort_clazz:
+            packages = sorted(packages, key=lambda p: p.name)
+
+        self._init_headers(packages)
+        self._init_insert_stmts(packages, sort_clazz)
+
+        self._print_headers()
+        self._print_get_entity_lookup_func()
+
+    def _init_headers(self, packages):
+        unique_headers = set()
+        unique_headers.add('#include "../src/top_entity_lookup.hpp"')
+        unique_headers.add('#include "../src/make_unique.hpp"')
+        unique_headers.add('#include "../src/entity.hpp"')
+        for package in packages:
+            header_name = package.get_cpp_header_name()
+            unique_headers.add('#include "%s"' % header_name)
+            for imported_type in package.imported_types():
+                header_name = imported_type.get_cpp_header_name()
+                unique_headers.add('#include "%s"' % header_name)
+
+        self.headers = list(sorted(unique_headers))
+
+    def _init_insert_stmts(self, packages, sort_clazz):
+        entity_lookup = {}
+        for package in packages:
+            top_level_entities = [entity
+                                  for entity in package.owned_elements
+                                  if hasattr(entity, 'stmt') and
+                                  entity.stmt.keyword in ('container', 'list')]
+            for top_entity in top_level_entities:
+                path = '/%s:%s' % (top_entity.module.arg, top_entity.stmt.arg)
+                entity_lookup[path] = top_entity.fully_qualified_cpp_name()
+
+        self.entity_lookup = entity_lookup
+
+    def _print_headers(self):
+        for header in self.headers:
+            self.ctx.writeln(header)
+
+    def _print_get_entity_lookup_func(self):
+        self.ctx.bline()
```

Comment:
Can this function be split into smaller functions per the clean code principles?